lib/statsd.js

```diff
@@ -62,6 +62,15 @@ function makeStatsD(options, logger) {
         statsd = new StatsD(statsdOptions);
     }
 
+    // Support creating sub-metric clients with a fixed prefix. This is useful
+    // for systematically categorizing metrics per-request, by using a
+    // specific logger.
+    statsd.makeChild = function(name) {
+        var childOptions = Object.assign({}, options);
+        childOptions._prefix = statsdOptions.prefix + normalizeName(name) + '.';
```

Comment:
Only `options.name` is ever normalised, `options._prefix` isn't; it's only used as-is. Also, what's up with the dot (cf. my previous comment) ? :P